### PR TITLE
Formatting fixes via 'qmk cformat'

### PR DESCRIPTION
**Pull request by xplusplus**
## Description

I was poking around the CLI code and noticed that there was a command to format c code. I ran it and several files were in need of clean up, so I decided to make a PR.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/quantum.c [hunk 1]
```diff
@@ -373,39 +373,39 @@ __attribute__((weak)) const bool ascii_to_altgr_lut[128] PROGMEM = {0, 0, 0, 0,
...
-                                                                         KC_GRV, KC_A, KC_B, KC_C, KC_D, KC_E, KC_F, KC_G,
-                                                                         // h     i        j        k        l        m        n        o
-                                                                         KC_H, KC_I, KC_J, KC_K, KC_L, KC_M, KC_N, KC_O,
-                                                                         // p     q        r        s        t        u        v        w
-                                                                         KC_P, KC_Q, KC_R, KC_S, KC_T, KC_U, KC_V, KC_W,
-                                                                         // x     y        z        {        |        }        ~        DEL
-                                                                         KC_X, KC_Y, KC_Z, KC_LBRC, KC_BSLS, KC_RBRC, KC_GRV, KC_DEL};
+__attribute__((weak)) const uint8_t ascii_to_keycode_lut[128] PROGMEM = {  // NUL   SOH      STX      ETX      EOT      ENQ      ACK      BEL
```
Comment (fauxpark) [inline quantum/quantum.c]: Could you instead wrap this in `// clang-format off`/`on` and align it like it is in the `keymap_extras/sendstring_*.h` files?

File: quantum/template/avr/config.h [hunk 2]
```diff
@@ -23,8 +23,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define VENDOR_ID 0xFEED
 #define PRODUCT_ID 0x0000
 #define DEVICE_VER 0x0001
-#define MANUFACTURER %YOUR_NAME%
-#define PRODUCT %KEYBOARD%
+#define MANUFACTURER % YOUR_NAME %
```
Comment (fauxpark) [inline quantum/template/avr/config.h]: Please undo the changes to the template files - clang-format adding the spaces here breaks the new_keyboard script. The LTO changes are OK though.

File: tests/basic/keymap.c [hunk 3]
```diff
@@ -22,7 +22,7 @@
 #define COMBO1 RSFT(LCTL(KC_O))
 
 const uint16_t PROGMEM
-               keymaps[][MATRIX_ROWS][MATRIX_COLS] =
+    keymaps[][MATRIX_ROWS][MATRIX_COLS] =
```
[the remark this answers is not in the document]
Comment (xplusplus) [reply tests/basic/keymap.c]: The clang formatter, both the QMK subcommand `qmk cformat`and the direct command `clang-format -style=file -i keymap.c` place this portion of the declaration on the next line.

Moving more of the code to a single line all the way to the opening curly brace results in this: 
image

I'll fix it manually.

File: quantum/quantum.c (hunk 1 above)
Comment (xplusplus) [reply quantum/quantum.c]: will do

File: quantum/template/avr/config.h (hunk 2 above)
Comment (xplusplus) [reply quantum/template/avr/config.h]: will do

File: tests/basic/keymap.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (fauxpark) [reply tests/basic/keymap.c]: Yeah, perhaps clang-format should be turned off for this bit too, considering it doesn't understand that keymaps are usually formatted to look like a keyboard.

File: keyboards/keebio/bdn9/config.h [hunk 4]
```diff
@@ -61,6 +61,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 /* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
 #define DEBOUNCE 5
 
+#define TAP_CODE_DELAY 10
```
Comment (xplusplus) [inline keyboards/keebio/bdn9/config.h]: Undo

File: keyboards/niu_mini/keymaps/foobar/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,103 @@
+#include QMK_KEYBOARD_H
```
Comment (xplusplus) [inline keyboards/niu_mini/keymaps/foobar/default/keymap.c]: undo

File: keyboards/niu_mini/keymaps/foobar/default/readme.md [hunk 6]
```diff
@@ -0,0 +1,2 @@
+# Default layout
```
Comment (xplusplus) [inline keyboards/niu_mini/keymaps/foobar/default/readme.md]: undo

**Comment by xplusplus** [comment]
I ended up going down a rabbit hole: many files turned off clang-format but didn't reenable it at the bottom so many files were missed by my initial pass. Also, the current clang-format config doesn't really jive with the existing format's people are using, so I don't think this is the right thing to do right now.

I think the best thing to do is close this and confer with the main contributors about perhaps updating the clang-format config to align with current coding conventions used in qmk_firmware. 

Thanks to all who contributed to this review. Closing
